src/test/lib/logical_query_plan/join_node_test.cpp

```diff
@@ -487,121 +487,197 @@ TEST_F(JoinNodeTest, FunctionalDependenciesUnify) {
    */
   const auto fds_unified = union_fds(non_trivial_fds, trivial_fds);
   EXPECT_EQ(fds_unified.size(), 3);
-  const auto fds_unified_set = std::unordered_set<FunctionalDependency>(fds_unified.cbegin(), fds_unified.cend());
-  EXPECT_TRUE(fds_unified_set.contains(expected_fd_a_b));
-  EXPECT_TRUE(fds_unified_set.contains(expected_fd_c));
-  EXPECT_TRUE(fds_unified_set.contains(expected_fd_x));
+  EXPECT_TRUE(fds_unified.contains(expected_fd_a_b));
+  EXPECT_TRUE(fds_unified.contains(expected_fd_c));
+  EXPECT_TRUE(fds_unified.contains(expected_fd_x));
   EXPECT_EQ(fds_unified, join_node->functional_dependencies());
 }
 
-TEST_F(JoinNodeTest, UniqueConstraintsSemiAndAntiJoins) {
-  _mock_node_a->set_key_constraints({*_key_constraint_a, *_key_constraint_b_c});
+TEST_P(JoinNodeMultiJoinModeTest, UniqueColumnCombinationsNoJoinColumnUnique) {
+  const auto join_mode = GetParam();
+  auto join_node = std::shared_ptr<JoinNode>{};
+  if (join_mode == JoinMode::Cross) {
+    join_node = JoinNode::make(join_mode, _mock_node_a, _mock_node_b);
+  } else {
+    // clang-format off
+    join_node =
+    JoinNode::make(join_mode, equals_(_t_a_a, _t_b_y),
+      _mock_node_a,
+      _mock_node_b);
+    // clang-format on
+  }
+
+  // Case 1 – LEFT  table's join column (a) uniqueness : No
+  //          RIGHT table's join column (y) uniqueness : No
```

Comment:
Since the different cases are not in a single test case anymore, adding some comments above the test cases to make it clear that they relate to each other might be helpful. This could be, for example, a combination table for the four different cases with a reference to the test cases.

Reply:
Added such a table: https://github.com/hyrise/hyrise/blob/7d18d5669ed497ee31e989e72d1a5614ec1e90ad/src/test/lib/logical_query_plan/join_node_test.cpp#L497